### PR TITLE
PUBDEV-3218: Add RMSLE to model metrics

### PR DESCRIPTION
**Pull request by navdeep-G**
This PR does the following:

- Add root mean squared log error to model metrics and CV summary for all regression models

- Add a method for extracting RMSLE from models in the R and Python API

- Modified `pyunit_metric_json_check.py` such that it extracts the newly added RMSLE. Otherwise it would fail.

- Add RMSLE as a sort metric for grid search

- Add RMSLE as a stopping metric for model building

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/98)

### CODE REVIEW

**Comment by abal5** [comment]
Can one of the admins verify this patch?

File: h2o-core/src/main/java/hex/ModelMetricsRegression.java [hunk 1]
```diff
@@ -85,6 +89,7 @@ static public ModelMetricsRegression make(Vec predicted, Vec actual, Distributio
     double _sumdeviance;
     Distribution _dist;
     double _abserror;
+    double _rmsleerror;
```
Comment (arnocandel) [inline h2o-core/src/main/java/hex/ModelMetricsRegression.java]: should be _rmslerror

File: h2o-core/src/main/java/hex/ScoreKeeper.java [hunk 2]
```diff
@@ -70,6 +71,7 @@ public void fillFrom(ModelMetrics m) {
     if (m instanceof ModelMetricsRegression) {
       _mean_residual_deviance = ((ModelMetricsRegression)m)._mean_residual_deviance;
       _mae = ((ModelMetricsRegression)m)._mean_absolute_error;
+      _rmsle = ((ModelMetricsRegression)m)._mean_absolute_error;
```
Comment (arnocandel) [inline h2o-core/src/main/java/hex/ScoreKeeper.java]: typo

**Comment by navdeep-G** [comment]
Fixed previous comments. See latest commit

**Comment by arnocandel** [comment]
1. Can we also have a unit test in R or python that tests that the values match for two large random vectors (actual/predicted), say 1e5 observations?

You can use make_metrics(predicted,actual) to compute the H2O version of RMSLE.


2. Also add tests for make_metrics() here:

https://github.com/h2oai/h2o-3/blob/c1986f1607a397b16b911f248987f7cf4abb93f9/h2o-py/tests/testdir_misc/pyunit_make_metrics.py

https://github.com/h2oai/h2o-3/blob/c1986f1607a397b16b911f248987f7cf4abb93f9/h2o-r/tests/testdir_misc/runit_make_metrics.R

**Comment by navdeep-G** [comment]
Added the above unit tests

File: h2o-r/tests/testdir_misc/runit_rmsle.R [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#Define some helper functions to get RMSLE manually
+
+# Compute the squared log error
+#
+# This function computes the elementwise squared log error for a
+# number or a vector
+#
+# @param actual ground truth number or vector
+# @param predicted predicted number or vector
+sle <- function (actual, predicted) (log(1+actual)-log(1+predicted))^2
```
Comment (arnocandel) [inline h2o-r/tests/testdir_misc/runit_rmsle.R]: I'd prefer the use of log1p(actual) over log(1+actual), but that's just cosmetics.

**Pull request merged**